### PR TITLE
Add support for Unsigned with CCValue

### PR DESCRIPTION
**Pull request by stevetranby**
This is meant to support getting and storing unsigned values into the Value class. The instigator of adding this is to fix a bug in TMX Tiled parser where flipped/rotated GIDs are not parsed correctly as they're larger than MAX_INT.

### CODE REVIEW

File: cocos/base/CCValue.cpp [hunk 1]
```diff
@@ -491,6 +522,50 @@ int Value::asInt() const
...
+
+    if (_type == Type::STRING)
+    {
+        // NOTE: strtoul is required (need to augment on unsupported platforms)
+        return static_cast<unsigned int>(strtoul(_field.strVal->c_str(), NULL, 10));
+    }
+
+    if (_type == Type::FLOAT)
+    {
+        return static_cast<unsigned int>(_field.floatVal);
```
Comment (minggo) [inline cocos/base/CCValue.cpp:553]: Here uses `static_cast<unsigned int>`, but it uses `static_cast<unsigned>` in other places. I think they both the same thing, so it is better to use only one style.

File: cocos/base/CCValue.cpp [hunk 2]
```diff
@@ -491,6 +522,50 @@ int Value::asInt() const
...
+
+    if (_type == Type::BYTE)
+    {
+        return static_cast<unsigned>(_field.byteVal);
+    }
+
+    if (_type == Type::STRING)
+    {
+        // NOTE: strtoul is required (need to augment on unsupported platforms)
+        return static_cast<unsigned int>(strtoul(_field.strVal->c_str(), NULL, 10));
```
Comment (minggo) [inline cocos/base/CCValue.cpp]: NULL -> nullptr?

File: cocos/base/CCValue.cpp (hunk 1 above)
Comment (stevetranby) [inline cocos/base/CCValue.cpp:553]: yep, thanks.

File: cocos/base/CCValue.cpp (hunk 2 above)
Comment (stevetranby) [inline cocos/base/CCValue.cpp]: yep, thanks.

**Comment by stevetranby** [comment]
Fixed based on notes.

Do you prefer me doing a rebase into a single commit and then force pushing, or do you recommend doing the merge to get the feature up-to-date?

Or maybe you prefer creating another pull request that includes only a single commit with all changes?

**Comment by minggo** [comment]
@stevetranby Thanks. Now github is easy to squash when merging a PR.

**Pull request merged**